saleor/warehouse/availability.py

```diff
@@ -61,6 +62,26 @@ def get_available_quantity_for_customer(
     return min(_get_available_quantity(stocks), settings.MAX_CHECKOUT_LINE_QUANTITY)
 
 
+def get_max_available_quantity_for_customer(variant: "ProductVariant") -> int:
+    """Return maximum checkout line quantity for all shipping zones."""
+    availability_in_shipping_zones = (
+        ShippingZone.objects.filter(warehouses__stock__product_variant=variant)
+        .values("pk")
```

Comment:
Why are you using `values("pk")` here?

Reply:
To group by `shipping_zone` on stock quantities.